config/environments/development.rb

```diff
@@ -55,4 +55,7 @@
 
   # Auto reload browser
   config.middleware.insert_before Rack::Lock, Rack::LiveReload if ENV['LIVE_RELOAD']
+
+  # Raise error on unpermitted strong params
+  config.action_controller.action_on_unpermitted_parameters = :raise
```

Comment:
@linesarefuzzy I am not in favour of this change. I guess this is why you put the `delete_if` in the strong params method. If so, please remove that also.

I understand the philosophical argument here, and I think I agree with it (without knowing, that is, the rationale of the Rails developers that did it the way they did--they may have had long debates on this exact point). But practically there are several issues here. 

I suspect this will start causing errors in other controllers that were not built with the expectation of errors on unpermitted params.

I understand that it is only in the development environment, but still. This project has a history of the development environment getting messed up and it is seriously jarring as a developer to never know if one is starting out with a clean environment or not. Development environment regressions are important too.

I understand that the less loud failure mode is annoying to you but I suggest you need to get in the habit of checking the logs as a first step when something is not working as it should when you are writing controller code. It is generally a good practice. I used to get tripped up by this too but now it has become second nature for me to check. We have many, many projects that use the default settings and changing them all to this setting is not an option. You will end up working on those codebases at some point so might as well learn now.